src/main/java/com/sapred/ordermanagerred/service/OrderService.java

```diff
@@ -53,5 +67,41 @@ else if (i % 3 == 1)
         orderRepository.saveAll(orders);
 
     }
+
+    public void fillProducts() {
+        for (int i = 1; i < 10; i++) {
+            AuditData d = new AuditData(new Date(), new Date());
+            List<Order> orders = new ArrayList<Order>();
+            ProductCategory pc = new ProductCategory(i + "", "name" + i, "desc" + i,
+                    companyRepository.findById("1").get(), new AuditData(new Date(), new Date()));
+            productCategoryRepository.save(pc);
+            Product p = new Product(i + "", "aaa", "aaa", 40, 50,
+                    DiscountType.PERCENTAGE, pc, 4,
+                    companyRepository.findById("1").get(), new AuditData(new Date(), new Date()));
+            productRepository.save(p);
+        }
+    }
+
+    public List<ProductCartDTO> calculateOrderAmount(Order order) {
```

Comment:
Code is nice

The problem is that once the user press "Add" the system performs a server call to calculate the amount for the newly added product and in your code, you calculate the amount for every ordered item which already calculated in the previous "Add"
So maybe you can calculate the order item only for item which amount is empty and not for all the items